clang/test/CodeGenSYCL/macro_definitions.cpp

```diff
@@ -0,0 +1,2 @@
+// RUN: %clang %s --sycl -sycl-std=1.2.1 -dM -E -x c++ | FileCheck %s
+// CHECK: #define CL_SYCL_LANGUAGE_VERSION 121
```

Comment:
By default the pre-processor tests are reside in https://github.com/intel/llvm/tree/sycl/clang/test/Preprocessor.
Also, I suggest using `-fsycl-is-device` option instead of `--sycl`.
We need to remove `--sycl` alias to avoid option duplicates.

Reply:
My understanding is that *-fsycl-is-device* only covers for when SYCL code is being build for the device, I need to test this flag also emits the macro when building SYCL code for the host. What would be the correct driver flag for that? *-fsycl*?

Reply:
> My understanding is that _-fsycl-is-device_ only covers for when SYCL code is being build for the device, I need to test this flag also emits the macro when building SYCL code for the host. What would be the correct driver flag for that? _-fsycl_?

Yes - `-fsycl`.